File: src/types/audio.ts
export interface AudioMetadata {
  path: string;
  duration: number | null;
  sample_rate: number | null;
  channels: number | null;
  bit_depth: number | null;
  codec: string | null;
  is_lossless: boolean;
  title: string | null;
  artist: string | null;
  album: string | null;
}

export type RepeatMode = "off" | "one" | "all";

export type PlaybackStatus = "stopped" | "playing" | "paused" | "loading";

export interface Track {
  path: string;
  name: string;
  metadata?: AudioMetadata;
}

File: src-tauri/src/models/audio.rs
use serde::Serialize;

#[derive(Debug, Clone, Serialize, Default)]
pub struct AudioMetadata {
    pub path: String,
    pub duration: Option<f64>,
    pub sample_rate: Option<u32>,
    pub channels: Option<u32>,
    pub bit_depth: Option<u32>,
    pub codec: Option<String>,
    pub is_lossless: bool,
    pub title: Option<String>,
    pub artist: Option<String>,
    pub album: Option<String>,
}
